### ISSUE TITLE
Try to detect multiple entries with the same ID

### ISSUE THREAD

**Issue by fluffy-critter** [fluffy-critter/Publ#39]
## Current Behavior

Sometimes it's possible to accidentally assign the same ID multiple times, like if you're working
on an entry in a branch and work on another entry in a different branch, and the IDs get generated on different machines with different existing indexes. 
## Possible Solution

One possible approach: when scanning an entry, if the entry record already exists and has a different pathname, see if the other pathname is still valid. If so, throw an error so it ends up in the log. Or something.

Another possibility: stop using autoincrement indexes, you silly

## Context

This seems unlikely to happen on real sites but I've run into it a few times while developing the example site content.

**Comment by fluffy-critter** [comment]
How about: scan entry, see if record exists and if so and the existing record’s path is valid and different, treat this entry as if it has no ID. Meaning it’ll get a new ID, possibly after being deferred into the fix up queue. (Obvs generate a warning in this case.)

Testing this will be obnoxious, and one possible result is permalinks getting reassigned, which is why having a better sequence generator in the first place is a good idea.

**Comment by fluffy-critter** [comment]
Possible improved sequence generator:

```
maxid = pow(10, .5 + log(entries.count()))
candidate = rand(maxid)
for entry in entries.select(id >= candidate):
    if entry.id > candidate:
        break
    candidate += 1
```

Idea being that it generates a random number with an upper bound based on the number of digits in the entry count (maybe it should use the max entry ID instead of the count?) and then increments it while it collides with an existing entry. Will have to do a quick test to see what the collision rate is with this approach.

**Comment by fluffy-critter** [comment]
(Also I like the idea of randomizing the sequence, as it makes it harder for people to guess about future entries. Great for comics etc)

**Comment by fluffy-critter** [comment]
Fixed in #42
